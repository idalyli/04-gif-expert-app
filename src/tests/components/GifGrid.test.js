import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");


jest.mock('../../hooks/useFetchGifs');



describe('Pruebas en el componente<GifGrid /> ', () => {

    const category = "hello";

    test('Debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({

            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    });

   test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const wrapper = shallow(<GifGrid category={category} />);

    })




})
