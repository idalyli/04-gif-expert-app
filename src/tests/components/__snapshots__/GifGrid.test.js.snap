// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente<GifGrid />  Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    hello
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
